polar-core/src/formatting.rs

```diff
@@ -430,12 +435,22 @@ pub mod to_polar {
                 ),
                 // Binary operators
                 Mul | Div | Add | Sub | Eq | Geq | Leq | Neq | Gt | Lt | Unify | Isa | In
-                | Assign => format!(
-                    "{} {} {}",
-                    to_polar_parens(self.operator, &self.args[0]),
-                    self.operator.to_polar(),
-                    to_polar_parens(self.operator, &self.args[1])
-                ),
+                | Assign => match self.args.len() {
```

Comment:
Can all of these operators appear in a three-arg formulation?

Reply:
Probably not `Unify`, `Isa`, `Assign`. I don't know that it's worth splitting them out, though.